src/types/commands.ts

```diff
@@ -684,6 +684,7 @@ export interface ActivateSheetCommand {
   type: "ACTIVATE_SHEET";
   sheetIdFrom: UID;
   sheetIdTo: UID;
+  autoSwitch?: boolean;
```

Comment:
It's not a good thing to add a flag in a widely used command only to handle a very specific use case in a very specific store. Also, the real issue is that `selectedMatchIndex` is reset in the method `refreshSearch`, and shouldn't in this use case. So, I suggest an alternative which is adding a flag **inside** the store:
```js
export class FindAndReplaceStore ...
...
private preserveSelectedMatchIndex: boolean = false;
...
finalize() {
// revert changes
}
...
private refreshSearch(jumpToMatchSheet = true) {
    if (!this.preserveSelectedMatchIndex) {
      this.selectedMatchIndex = null;
    }
    this.findMatches();
    this.selectNextCell(Direction.current, jumpToMatchSheet);
  }
...
if (jumpToMatchSheet && this.getters.getActiveSheetId() !== selectedMatch.sheetId) {
      this.preserveSelectedMatchIndex = true;
      this.model.dispatch("ACTIVATE_SHEET", {
        sheetIdFrom: this.getters.getActiveSheetId(),
        sheetIdTo: selectedMatch.sheetId,
        autoSwitch: true,
      });
      this.preserveSelectedMatchIndex = false;
      // We do not want to reset the selection at finalize in this case
      this.isSearchDirty = false;
    }
...
```

All tests are green with theses changes, the diff is smaller and the logic is kept inside the store and do not leak elsewhere.

Side note, `autoSwitch` and `isAutoSheetSwitch` are not really meaningful, it's important (and really difficult :D ) to have good names, it's easier to read the code (not only now, but in 2 or 3 years :) )